src/nhs_grid.jl

```diff
@@ -51,25 +61,44 @@ since not sorting makes our implementation a lot faster (although less paralleli
   In: Computer Graphics Forum 30.1 (2011), pages 99–112.
   [doi: 10.1111/J.1467-8659.2010.01832.X](https://doi.org/10.1111/J.1467-8659.2010.01832.X)
 """
-struct GridNeighborhoodSearch{NDIMS, ELTYPE, CL, PB, UB} <: AbstractNeighborhoodSearch
-    cell_list       :: CL
-    search_radius   :: ELTYPE
-    periodic_box    :: PB
-    n_cells         :: NTuple{NDIMS, Int}    # Required to calculate periodic cell index
-    cell_size       :: NTuple{NDIMS, ELTYPE} # Required to calculate cell index
-    update_buffer   :: UB # Multithreaded buffer for `update!`
-    threaded_update :: Bool
+struct GridNeighborhoodSearch{NDIMS, update_strategy,
+                              ELTYPE, CL, PB, UB} <: AbstractNeighborhoodSearch
+    # Store `update_strategy` as type parameter to dispatch `update!`
```

Comment:
I know that @sloede is not a fan of storing things as type parameters. I find it quite elegant that we can dispatch on the update strategy.

Reply:
I am not against storing things as type parameters if they can be used for dispatch. It's just that in my mind, `snake_case` names are strongly associated with variable and function names, which is why I find it somewhat weird. But I think it makes sense here.